### PR TITLE
Deprecate Locator.autoscale.

### PR DESCRIPTION
**Pull request by anntzer**
Note that this likely indicates that the dates Locators need new
implementations of view_limits, as they appear to never have been
updated to use the "new" (post-2008) API.

per https://github.com/matplotlib/matplotlib/issues/13292#issuecomment-464855904.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by timhoffm** [comment]
Both `Locator.view_limits()` and `Axis.get_view_interval()` could need some more documentation.

Actually, I have no clear idea what `view_limits()` is actually doing, which comes part due to a missing verb in the method name (set/get/update/calculate,...) part due to a lack of documentation. Is it a setter/a getter; does it calculate something; does it modify the internal state; what does it return?

**Comment by anntzer** [comment]
It expands the axes limits when using the old "round numbers" mode (still rcparams-switchable, where the axis would be expanded up to the closest surrounding ticks (so e.g. 1.1..4.1 would get expanded to 0..5) (which is the whole reason the locators have to get involved in the autoscaling) (nowadays the default is to just apply constant margins, of course).

Certainly it could be better documented, though that's not the object of this PR.

File: lib/matplotlib/dates.py [hunk 1]
```diff
@@ -1304,6 +1304,7 @@ def get_unit_generic(freq):
     def _get_interval(self):
         return self.rule._rrule._interval
 
+    @cbook.deprecated("3.1")
```
Comment (timhoffm) [inline lib/matplotlib/dates.py]: Please update all deprecations to 3.2. You'll get an approval as a reward. :wink:

Comment (anntzer) [reply lib/matplotlib/dates.py]: done

**Pull request merged**